packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-file-beta/gux-form-field-file.tsx

```diff
@@ -0,0 +1,368 @@
+    this.disabled = this.input.disabled;
+    this.required = this.input.required;
+
+    this.disabledObserver = onDisabledChange(
+      this.input,
+      (disabled: boolean) => {
+        this.disabled = disabled;
+      }
+    );
+    this.requiredObserver = onRequiredChange(
+      this.input,
+      (required: boolean) => {
+        this.required = required;
+      }
+    );
+    this.multipleObserver = onMultipleChange(this.input, () => {
+      forceUpdate(this.root);
+    });
+
+    validateFormIds(this.root, this.input);
+  }
+
+  private getDropZoneText(): string {
+    if (this.input.multiple) {
+      return this.getI18nValue('dragAndDropFilesInstructions');
+    } else {
+      return this.getI18nValue('dragAndDropFileInstructions');
+    }
+  }
+
+  private getProxyButtonText(): string {
+    if (this.dragAndDrop) {
+      return this.getI18nValue('clickToUpload');
+    } else if (this.input.multiple) {
+      if (this.input.files.length > 0) {
+        return this.getI18nValue('changeFiles');
+      }
+
+      return this.getI18nValue('uploadFiles');
+    } else {
+      if (this.input.files.length > 0) {
+        return this.getI18nValue('changeFile');
+      }
+      return this.getI18nValue('uploadFile');
+    }
+  }
+
+  private removeFile(index: number): void {
+    const fileList = this.input.files;
+    const dt = new DataTransfer();
+
+    for (let i = 0; i < fileList.length; i++) {
+      if (index !== i) {
+        dt.items.add(fileList[i]);
+      }
+    }
+
+    this.input.files = dt.files;
+    forceUpdate(this.root);
+  }
+
+  private onProxyFileButtonClick(event: MouseEvent): void {
+    event.preventDefault();
+    event.stopImmediatePropagation();
+
+    this.input.focus();
+    this.input.click();
+  }
+
+  private onDrop(event: DragEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (this.disabled) {
+      return;
+    }
+
+    const currentFileList = this.input.files;
+    const newFileList = event.dataTransfer.items;
+
+    const dt = new DataTransfer();
+
+    for (let i = 0; i < currentFileList.length; i++) {
+      dt.items.add(currentFileList[i]);
+    }
+
+    for (let i = 0; i < newFileList.length; i++) {
+      const item = newFileList[i];
+
+      if (item.kind === 'file' && item.webkitGetAsEntry().isFile) {
+        console.log(item, item.webkitGetAsEntry());
+        dt.items.add(item.getAsFile());
+      }
+    }
+
+    this.input.files = dt.files;
+    this.dropContainer.classList.remove('gux-drag-over');
+    forceUpdate(this.root);
+  }
+
+  private onDragOver(event: DragEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (!this.disabled) {
+      this.dropContainer.classList.add('gux-drag-over');
+    }
+  }
+
+  private onDragEnter(event: DragEvent): void {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (!this.disabled) {
+      this.dropContainer.classList.add('gux-drag-over');
+    }
+  }
```

Comment:
I think this could just be one function with a neutral name maybe but is fine as is.

Reply:
Done